src/metatrain/utils/transfer.py

```diff
@@ -23,9 +23,12 @@ def batch_to(
     :param dtype: Desired floating point data type.
     """
 
-    systems = [system.to(dtype=dtype, device=device) for system in systems]
+    systems = [
+        system.to(dtype=dtype, device=device, non_blocking=True) for system in systems
```

Comment:
From https://docs.pytorch.org/tutorials/intermediate/pinmem_nonblock.html, it seems that we might want to only do non_blocking when moving from CPU to GPU? 

> While `cpu_tensor.to("cuda", non_blocking=True).mean()` executes correctly, attempting `cuda_tensor.to("cpu", non_blocking=True).mean()` will result in erroneous outputs.

Or we should do something to prevent the issues mentioned there.